### ISSUE TITLE
Allow case-insensitive search

### ISSUE THREAD

**Issue by nkakouros** [vit-project/vit#208]
This is common in applications like `vim`, `less`, etc.

**Comment by thehunmonkgroup** [comment]
I'm in favor of the idea, and, we'd need to figure out how best to implement it:

 * VIT's default structure is vim-like
 * Among other things, vim allows that via ```\c```, ```\C```, and the ```ignorecase``` and ```smartcase``` settings
 * However, VIT is designed to support other keyboard configurations, so my preference is to avoid hard-coding things to expect the vim approach

Thoughts?

**Comment by scottkosty** [comment]
I agree it would be nice to have. I'm not sure what the best interface is.

**Comment by ahillio** [comment]
It'd be really sweet to have vim's `smartcase` setting :)

**Comment by nfraprado** [comment]
I think `ignorecase` and `smartcase` configurations are a must-have.
Being able to override this configuration for a single search also seems useful. I don't really like the `\c` and `\C` solution, which would require the search text to be parsed for this special sequence, but I couldn't think of a better one either.
Other options would be having commands for case-insensitive searches (`SEARCH_FORWARD_INSENSITIVE`, `SEARCH_REVERSE_INSENSITIVE`, or for NEXT and PREVIOUS), or a command that sets case-insensitivity until it is disabled. But `\c` and `\C` still seem a better option.

I couldn't resist implementing the `ignorecase` and `smartcase` configurations [here](https://github.com/nfraprado/vit/commit/4d65bc85f74ceae99c9f8922125e8382c8b93a75). It got me thinking if instead of these two separate configurations, which are really three possible values, couldn't it be a single `casemethod` configuration with `sensitive`, `insensitive` and `smart` as possible values. It would eliminate the weird case where `ignorecase` and `smartcase` can both be used in the configuration.
